py-polars/polars/io/spreadsheet/functions.py

```diff
@@ -738,13 +747,49 @@ def _read_spreadsheet_openpyxl(
         {s.name: s for s in series_data},
         schema_overrides=schema_overrides,
     )
-    if raise_if_empty and len(df) == 0 and len(df.columns) == 0:
-        msg = (
-            "empty Excel sheet"
-            "\n\nIf you want to read this as an empty DataFrame, set `raise_if_empty=False`."
-        )
-        raise NoDataError(msg)
-    return _drop_unnamed_null_columns(df)
+    return _drop_null_data(df, raise_if_empty=raise_if_empty)
+
+
+def _read_spreadsheet_calamine(
+    parser: Any,
+    sheet_name: str | None,
+    read_csv_options: dict[str, Any] | None,
+    schema_overrides: SchemaDict | None,
+    *,
+    raise_if_empty: bool,
+) -> pl.DataFrame:
+    ws = parser.load_sheet_by_name(sheet_name)
```

Comment:
NOTE: Once calamine 0.24.0 is out and https://github.com/ToucanToco/fastexcel/pull/147 is polished and merged, you could use one of the new eager loading methods to lower the memory footprint here